Mathlib/RingTheory/Length.lean

```diff
@@ -156,7 +156,7 @@ lemma Module.length_eq_add_of_exact :
         t.map ⟨Submodule.map f, Submodule.map_covBy_of_injective hf⟩
       have hfg : Submodule.map f ⊤ = Submodule.comap g ⊥ := by
         rw [Submodule.map_top, Submodule.comap_bot, LinearMap.exact_iff.mp H]
-      let r := t'.smash s' (by simpa [s', t', hs₁, ht₂])
+      let r := t'.smash s' (by simpa [s', t', hs₁, ht₂] using hfg)
```

Comment:
What happened here?

Reply:
I think the new lemmas resulted in simp skipping past the available assumptions